API/API.mli

```diff
@@ -5666,7 +5666,7 @@ module Vernacinterp :
 sig
   type deprecation = bool
 
-  type vernac_command = Genarg.raw_generic_argument list -> unit -> unit
+  type vernac_command = Genarg.raw_generic_argument list -> Loc.t option -> unit
```

Comment:
Maybe this should be `Genarg.raw_generic_argument list Loc.located` ?

Reply:
@matejkosik Can you take @ejgallego's comment into account and update this PR? Thanks!

Reply:
Do I interpret the suggestion of @ejgallego correctly? Do we really want to change the types of all variables bound in `VERNAC COMMAND EXTEND` macro clauses?

(In other words, we would break all plugins which use this macro.)

Reply:
Ok, after talking offline I think there was a misunderstanding. @ejgallego suggested to move to `Genarg.raw_generic_argument list Loc.located`, not `Genarg.raw_generic_argument located list`, which was what @matejkosik didn't know how to do.

Reply:
Yeah, anyways I dunno what makes sense here, [certainly breaking the API makes no sense]. At this level we tend to have two kind of objects: objects that always carry a location, and objects that may carry a location. I would be cool to handle them with different types, we got quite a bit of space to improve the API here.